bioblend/_tests/TestGalaxyFolders.py

```diff
@@ -36,6 +39,55 @@ def test_show_folder_contents(self):
         assert "metadata" in f2
         assert self.name == f2["metadata"]["folder_name"]
 
+    @test_util.skip_unless_galaxy("release_21.05")
+    def test_show_folder_contents_limit(self):
+        for i in range(12):
+            self.gi.folders.create_folder(self.folder["id"], f"{self.name} {i}")
+
+        # check defaults for limit and offset
+        f2 = self.gi.folders.show_folder(self.folder["id"], contents=True)
+        assert len(f2["folder_contents"]) == 10
+        assert f2["folder_contents"][0]["name"] == f"{self.name} 0"
+
+        # check non defaults
+        f2 = self.gi.folders.show_folder(self.folder["id"], contents=True, limit=1, offset=1)
+        assert len(f2["folder_contents"]) == 1
+        assert f2["folder_contents"][0]["name"] == f"{self.name} 1"
+
+    def test_show_folder_contents_include_deleted(self):
+        history = self.gi.histories.create_history(name="Test History")
+        hda_id = self._test_dataset(history["id"])
+
+        # Create 2 library datasets into the library folder
+        ldda1 = self.gi.libraries.copy_from_dataset(
+            library_id=self.library["id"], dataset_id=hda_id, folder_id=self.folder["id"], message="Added HDA"
+        )
+        ldda2 = self.gi.libraries.copy_from_dataset(
+            library_id=self.library["id"], dataset_id=hda_id, folder_id=self.folder["id"], message="Added HDA"
+        )
+        folder_info = self.gi.folders.show_folder(self.folder["id"], contents=True)
+        assert len(folder_info["folder_contents"]) == 2
+        assert folder_info["folder_contents"][0]["type"] == "file"
+
+        # Delete the library datasets and check if include_deleted works
+        self.gi.libraries.delete_library_dataset(self.library["id"], ldda1["id"])
+        self.gi.libraries.delete_library_dataset(self.library["id"], ldda2["id"], purged=True)
+        folder_info = self.gi.folders.show_folder(self.folder["id"], contents=True, include_deleted=True)
+        # check if there are 2 contents and the number is correct
+        assert len(folder_info["folder_contents"]) == 2
+        assert folder_info["metadata"]["total_rows"] == 2
+        # if called without contents=True an item_count is reported .. lets also check this
+        folder_info = self.gi.folders.show_folder(self.folder["id"], include_deleted=True)
+        assert folder_info["item_count"] == 2
+
+        folder_info = self.gi.folders.show_folder(self.folder["id"], contents=True)
+        assert len(folder_info["folder_contents"]) == 0
+        assert folder_info["metadata"]["total_rows"] == 0
```

Comment:
`total_rows` in the metadata (which is only returned if `contents=True`) seems correct .. but slightly annoying that the key is different.